### PR TITLE
Initial PoC of libnetwork in docker.

### PR DESCRIPTION
**Pull request by tomwilkie**

### CODE REVIEW

**Comment by tomwilkie** [comment]
Work in progress.

File: daemon/networkdriver/portmapper/proxy.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ import (
 const userlandProxyCommandName = "docker-proxy"
 
 func init() {
-	reexec.Register(userlandProxyCommandName, execProxy)
+	//reexec.Register(userlandProxyCommandName, execProxy)
```
Comment (squaremo) [inline daemon/networkdriver/portmapper/proxy.go:23]: Why?

Comment (tomwilkie) [inline daemon/networkdriver/portmapper/proxy.go:23]: This has been moved into libnetwork now - and reexec.Register complains (cannot be done twice.)

Comment (squaremo) [inline daemon/networkdriver/portmapper/proxy.go:23]: Got it, ta

**Comment by tomwilkie** [comment]
This is no longer WIP; good to merge?

**Comment by squaremo** [comment]
> good to merge?

Yes, though next we ought to remove the conditionals and just implement using libnetwork; we can keep our own implementation in another branch.

**Comment by tomwilkie** [comment]
Okay I'll have a go at that; will merge this in the mean time.

**Pull request merged**